snow/validators/test_retriever.go

```diff
@@ -0,0 +1,26 @@
+package validators
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/flare-foundation/flare/ids"
+)
+
+var (
+	errGetValidatorsByBlockID = errors.New("unexpectedly called GetValidatorsByBlockID")
+)
+
+type TestRetriever struct {
+	T *testing.T
+
+	GetValidatorsByBlockIDF func(blockID ids.ID) (Set, error)
+}
+
+func (T *TestRetriever) GetValidatorsByBlockID(blockID ids.ID) (Set, error) {
+	if T.GetValidatorsByBlockIDF != nil {
+		return T.GetValidatorsByBlockIDF(blockID)
+	}
+
+	return nil, errGetValidatorsByBlockID
+}
```

Comment:
I'd recommend instead just simplifying it to a [simple mock](https://optakt.github.io/code/testing/#mocks) as described in our testing guide.

The mock's func should always be set so that the behavior of the mock is always explicit and clear when reading the test. If there are tons of cases where we need this func call to just not error and it's useless to repeat the same func over and over again, I'd suggest having a mock and a function that builds a BaselineMock as described in the knowledge base as well

```suggestion
type RetrieverMock struct {
	GetValidatorsByBlockIDFunc func(blockID ids.ID) (Set, error)
}

func (m *RetrieverMock) GetValidatorsByBlockID(blockID ids.ID) (Set, error) {
		return m.GetValidatorsByBlockIDF(blockID)
}
```

Also it's very unusual to have an uppercase pointer receiver name 😄